Convert md files to rst, split CONTRIBUTING.md (#5)
ansible/community-docs: create_pr_quick_start_guide.rst

*****************************

> Quick-start development guide

Reviewer: Since this guide is about collections, maybe we should drop the ansible/ansible git stuff because it is not strictly necessary? Using the latest stable Ansible Core should be good enough. This will also help keep the guide a bit more focused.

I would also introduce venv here to keep the things contained and easy to remove later on.

Author: @tadeboro , thanks for the suggestions!
1)  Ansible-Core - removed
2) venv - i'd leave only really necessary things there to make the guide really quick:)

*****************************

> This quide describes all steps needed to create your first patch and submit a pull request.

Reviewer 2: ```suggestion
This guide describes all steps needed to create your first patch and submit a pull request.
```

7. Go to your new cloned repository.

.. code:: bash

  cd ~/ansible_collections/NAMESPACE/COLLECTION_NAME

8. Be sure you are in the default branch (it is usually ``main``):

.. code:: bash

  git status

9. Show remotes. There should be the ``origin`` repository only:

.. code:: bash

  git remote -v

10. Add the ``upstream`` repository:

.. code:: bash

  git remote add upstream https://github.com/ansible-collections/COLLECTION_REPO.git

This is the repository where you forked from.

11. Update your local default branch. Assuming that it is ``main``:

.. code:: bash

  git fetch upstream
  git rebase upstream/main

12. Create a branch for your changes:

.. code:: bash

  git checkout -b name_of_my_branch

13. We recommend you start with writing integration tests if applicable.

Note: If there are any difficulties with writing / running the tests or you are not sure if the case can be covered, feel free to skip this step.
If needed, other contributors can help you with it later.

Note: Some collections do not have integration tests.

All integration tests are stored in ``tests/integration/targets`` subdirectories.

> Go to the subdirectory containing the name of module you are going to change.

Reviewer 2: ```suggestion
Go to the subdirectory containing the name of the module you are going to change.
```

.. code:: bash

  ansible-test integration name_of_test_subdirectory --docker -v

For example, if the tests files you changed are stored in ``tests/integration/targets/test_mysql_user/``, the command will be:

.. code:: bash

  ansible-test integration test_mysql_user --docker -v

You can use the ``-vv`` or ``-vvv`` argument, if you need more detailed output.

In the examples above, the default test image will be automatically downloaded and used to create and run a test container.
Use the default test image for platform independent integration tests such as those for cloud modules.

If you need to run the tests against a specific distribution, see the `list of supported container images <https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#container-images>`_. In this case, the command can look like:

.. code:: bash

  ansible-test integration name_of_test_subdirectory --docker centos8 -v

Note: If you are not sure whether you should use the default image for testing or a specific one, skip the entire step - the community will help you later.
You can also try to use the collection repository's CI to figure out which containers are used.

> If the tests do not want to run, first, check you complete step 3 of this guide.

Reviewer 2: ```suggestion
If the tests do not want to run, first, check that you completed step 3 of this guide.
```

If the tests ran successfully, there are usually two possible outcomes:

> a) If the bug has not appeared and the tests have passed successfully, ask the reporter to provide more details. The bug can be not a bug actually or can relate to a particular software version used or specifics of local environment configuration.

Reviewer 2: ```suggestion
a) If the bug has not appeared and the tests have passed successfully, ask the reporter to provide more details. The bug can be not a bug actually or can relate to a particular software version used or specifics of the local environment configuration.
```

> b) The bug has appeared and the tests has failed as expected showing the reported symptoms.

Reviewer 2: I'm not sure why, but the a), b) here isn't showing up well if you view the file in preview mode. Here and the a/b below as well. Possibly it just needs a newline before the a) in each?

Author: Should be fixed, thanks for pointing out!

For example, if the tests files you changed are stored in ``tests/integration/targets/test_mysql_user/``, the command will be:

.. code:: bash

  ansible-test integration test_mysql_user --docker -v

You can use the ``-vv`` or ``-vvv`` argument, if you need more detailed output.

If you need to run the tests against a specific distribution, see step 13.

There are two possible outcomes:
a) They have failed. Look at the output of the command.
Fix the problem place in the code and run again.
Repeat the cycle until the tests pass.

b) They have passed. Remember they failed originally? Our congratulations! You have fixed the bug.

18. Commit your changes with an informative but short commit message:

.. code:: bash

  git add /path/to/changed/file
  git commit -m "module_name_you_fixed: fix crash when ..."

19. Push the branch to the ``origin`` (your fork):

.. code:: bash

  git push origin name_of_my_branch

20. Go to the ``upstream`` (http://github.com/ansible-collections/COLLECTION_REPO).

21. Go to ``Pull requests`` tab and create a pull request.

GitHub is tracking your fork, so it should see the new branch in it and automatically offer

> to create a pull request. Sometimes GitHub does not do it and you should click the ``New pull request`` button yourself.

Reviewer 2: ```suggestion
to create a pull request. Sometimes GitHub does not do it, and you should click the ``New pull request`` button yourself.
```
